src/views/tickets/Yara.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import { Segment, List } from 'semantic-ui-react';
+import { getArtifactListing, getArtifact, getTicket } from '../../utils/api.js';
+
+const REFRESH_EVERY_MS = 1000;
+
+export default class Yara extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      processed: false,
+      error: false,
+      isMalicous: false,
+      matchedRules: [],
+    };
+    this.processed = false;
+
+    this.refreshInterval = setInterval(
+      this.refreshTicketState,
+      REFRESH_EVERY_MS
+    );
+
+    this.refreshYaraState();
+  }
+
+  stopAutomaticRefreshing() {
+    clearInterval(this.refreshInterval);
+  }
+
+  async fetchArtifacts() {
+    const ticketId = this.props.ticketID;
+
+    const res = await getArtifactListing(ticketId);
+    const artifacts = res.fileArtifacts.filter(artifact =>
+      artifact.filename.endsWith('.yara')
+    );
+
+    let yara = [];
+
+    for (const item of artifacts) {
+      let artifact = await getArtifact(this.props.ticketID, item.filename);
+      yara.push(artifact);
+    }
+    return yara;
+  }
+
+  refreshYaraState = async e => {
+    try {
+      const ticketId = this.props.ticketID;
+      const ticket = await getTicket(ticketId);
+      const { processed } = ticket.ticket;
+
+      let artifacts = await this.fetchArtifacts();
+      let matchedRules = [];
+
+      let i = 0;
+      for (i; i < artifacts.length; i++) {
+        let item = JSON.parse(artifacts[i]);
+        const rule = JSON.parse(item.matchedRule.substr(7));
+        console.log('This is the information that will need to be printed');
+        console.log(rule.rules[0]);
+        matchedRules.push([i, rule.rules[0], item.error]);
+      }
+
+      this.setState({
+        processed: processed,
```

Comment:
nit: `this.setState({ processed, matchedRules });`

Reply:
Is this something that still needs to be changed?  If so, could you please elaborate?  I'm not really sure what this comment is referencing

Reply:
Sorry! If the name of the variable is the same as the name of the key, you can omit the value. So if a variable in-scope is named `foo` (e.g. `let foo = 1;`, you can include it in an object by saying `let bar = { foo };` instead of `let bar = { foo: foo };`). I start comments with nit: if it's something that's a style change, and doesn't need to be fixed.

Reply:
Oh ok, sounds good, thanks for the help, I just didn't know what the comment was, I got it now.

Reply:
👍